### PR TITLE
Multiple crash cherrypick

### PR DESCRIPTION
**Pull request by pinzart90**

### CODE REVIEW

File: test/DynamoCoreWpfTests/SplashScreenTests.cs [hunk 1]
```diff
@@ -1,13 +1,34 @@
...
+        }
+
+        [DllImport("user32.dll")]
+        public static extern IntPtr FindWindow(string lpClassName, String lpWindowName);
+
+        [DllImport("user32.dll")]
+        public static extern int SendMessage(IntPtr hWnd, int wMsg, IntPtr wParam, IntPtr lParam);
+
+        [DllImport("user32.dll", SetLastError = true, CharSet = CharSet.Auto)]
+        static extern uint RegisterWindowMessage(string lpString);
```
Comment (mjkkirschner) [inline test/DynamoCoreWpfTests/SplashScreenTests.cs:29]: is this used?

File: test/DynamoCoreWpfTests/SplashScreenTests.cs [hunk 2]
```diff
@@ -1,13 +1,34 @@
...
     [TestFixture]
     internal class SplashScreenTests
     {
+        public enum WindowsMessage
+        {
+            WM_CLOSE = 0x0010
+        }
+
+        [DllImport("user32.dll")]
+        public static extern IntPtr FindWindow(string lpClassName, String lpWindowName);
```
Comment (mjkkirschner) [inline test/DynamoCoreWpfTests/SplashScreenTests.cs:23]: Any reason you can't use WPF tree for this?

**Comment by mjkkirschner** [comment]
Are there two crash fixes in here or just one?

**Review by mjkkirschner — APPROVED** [review]

**Comment by mjkkirschner** [comment]
LGTM, but I have no idea if 3.0.3 is open or not.

**Comment by pinzart90** [comment]
> hey @pinzart90 I know this is a cherry pick but I must have missed the original test functions - why depend on those instead of using wpf? Just a question, not really a review comment.

One crash fix

File: test/DynamoCoreWpfTests/SplashScreenTests.cs (hunk 2 above)
Comment (pinzart90) [reply test/DynamoCoreWpfTests/SplashScreenTests.cs:23]: The issue does not reproduce if the window is closed from managed code. THere are different code paths taken

File: test/DynamoCoreWpfTests/SplashScreenTests.cs (hunk 1 above)
Comment (pinzart90) [reply test/DynamoCoreWpfTests/SplashScreenTests.cs:29]: No, it was during my first pass of writing the test.
I'll clean it up in master

**Pull request closed without merging**